x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -2075,4 +2083,22 @@ public void close() {
             }
         }
     }
+
+    private static String getFileExtension(String resourceDescription) {
+        // TODO: consider introspecting resourceDescription for compound files
+        if (resourceDescription.endsWith(LuceneFilesExtensions.CFS.getExtension())) {
+            return LuceneFilesExtensions.CFS.getExtension();
+        }
+        try {
+            String extension = IndexFileNames.getExtension(resourceDescription);
+            if (LuceneFilesExtensions.isLuceneExtension(extension)) {
```

Comment:
should we add some unit tests for `isLuceneExtension` ? (lucene, non-lucene values, nulls etc)